dm/worker/server.go

```diff
@@ -61,7 +61,7 @@ type Server struct {
 	// closeMu is used to sync Start/Close and protect 5 fields below
 	closeMu    sync.Mutex
 	closed     atomic.Bool
-	inited     bool
+	callClosed bool
```

Comment:
```suggestion
	calledClose bool
```
plz add some comments to explain the difference between `closed` and `calledClose`